schemas/src/test/java/com/scottlogic/deg/schemas/v0_1/ProfileSchemaValidatorTests.java

```diff
@@ -0,0 +1,82 @@
+import org.junit.jupiter.api.DynamicTest;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.Supplier;
+
+public class ProfileSchemaValidatorTests {
+
+    private final String TEST_PROFILE_DIR = "/test-profiles/";
+    private final String INVALID_PROFILE_DIR = "invalid";
+    private final String VALID_PROFILE_DIR = "valid";
+
+    FilenameFilter jsonFilter = new FilenameFilter() {
+        public boolean accept(File dir, String name) {
+            String lowercaseName = name.toLowerCase();
+            if (lowercaseName.endsWith(".json")) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    };
+
+    private File getFileFromURL(String profileDirName) {
+        URL url = this.getClass().getResource(TEST_PROFILE_DIR + profileDirName);
+        File file = null;
+        try {
+            file = new File(url.toURI());
+        } catch (URISyntaxException e) {
+            file = new File(url.getPath());
+        } finally {
+            return file;
+        }
+    }
+
+    Collection<DynamicTest> testInvalidProfiles(ProfileSchemaValidator profileValidator) {
+        File[] listOfFiles = getFileFromURL(INVALID_PROFILE_DIR).listFiles(jsonFilter);
+        Collection<DynamicTest> dynTsts = new ArrayList<DynamicTest>();
+
+        for (int i = 0; i < listOfFiles.length; i++) {
+            String profileFilename = listOfFiles[i].getName();
+            DynamicTest test = DynamicTest.dynamicTest(profileFilename, () -> {
+                //    System.out.println("Testing " + INVALID_PROFILE_DIR + " profile [" + profileFilename + "]");
+                InputStream testProfile = this.getClass().getResourceAsStream(TEST_PROFILE_DIR + INVALID_PROFILE_DIR + File.separator + profileFilename);
+                ValidationResult result = profileValidator.validateProfile(testProfile);
+                Supplier<String> msgSupplier = () -> "Profile ["
+                    + profileFilename + "] should not be valid";
+                Assertions.assertFalse(result.isValid(), msgSupplier);
+            });
+            dynTsts.add(test);
+        }
+        return dynTsts;
+    }
+
+    Collection<DynamicTest> testValidProfiles(ProfileSchemaValidator profileValidator) {
+        File[] listOfFiles = getFileFromURL(VALID_PROFILE_DIR).listFiles(jsonFilter);
+        Collection<DynamicTest> dynTsts = new ArrayList<DynamicTest>();
+
+        for (int i = 0; i < listOfFiles.length; i++) {
+            String profileFilename = listOfFiles[i].getName();
+            DynamicTest test = DynamicTest.dynamicTest(profileFilename, () -> {
+                //      System.out.println("Testing " + VALID_PROFILE_DIR + " profile [" + TEST_PROFILE_DIR + VALID_PROFILE_DIR  + File.separator+  profileFilename + "]");
```

Comment:
The `DynamicTest` includes the name of the profile; is there a need for this anymore?

Reply:
What else would you want to put in the test?

Reply:
yes, removed the `println` statement now